### PR TITLE
Ensure DID Controller is used consistently. Fix #37.

### LINKED ISSUE #37: Add "DID controller" to "Terminology" section.
@peacekeeper moved from CCG (https://github.com/w3c-ccg/did-spec/issues/205)

### PR DESCRIPTION
**Pull request by msporny**
***
<a href="https://pr-preview.s3.amazonaws.com/w3c/did-spec/pull/77.html" title="Last updated on Oct 23, 2019, 1:23 AM UTC (df93787)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/did-spec/77/99a0873...df93787.html" title="Last updated on Oct 23, 2019, 1:23 AM UTC (df93787)">Diff</a>

### CODE REVIEW

**Comment by msporny** [comment]
> At some point there were discussions about the difference between "controller of a DID" and "controller of a DID document".

That's true, and we still need to deal with that... waiting on @jandrieu to provide some guidance there as I thought he had the beginnings of a plan the last time this came up.

**Comment by peacekeeper** [comment]
Also, in some cases this PR changes plural language "controller(s) of a DID" to singular "DID controller".. Is that a problem?

**Comment by TallTed** [comment]
> Also, in some cases this PR changes plural language "controller(s) of a DID" to singular "DID controller".. Is that a problem?

I think, yes, because it will suggest to some readers that there can only be a single controller, which interpretation was intentionally avoided by including the `(s)`.

File: index.html [hunk 1]
```diff
@@ -1367,14 +1367,14 @@ <h2>
       </h2>
 
       <p>
-Authentication is the mechanism by which the controller(s) of a DID can
+Authentication is the mechanism by which the <a>DID controller</a> can
```
Comment (TallTed) [inline index.html]:
```suggestion
Authentication is the mechanism by which the <a>DID controller(s)</a> can
```

File: index.html [hunk 2]
```diff
@@ -1367,14 +1367,14 @@ <h2>
...
 cryptographically prove that they are associated with that DID.
 See Section <a href="#binding-of-identity"></a>. Note
-that Authentication is separate from Authorization because the controllers
-may wish to enable others to update their DID Document (for
-example, to assist with key recovery as discussed in Section
+that Authentication is separate from Authorization because the
+<a>DID controller</a> may wish to enable others to update their DID Document
```
Comment (TallTed) [inline index.html]:
```suggestion
<a>DID controller(s)</a> may wish to enable others to update their DID Document
```

File: index.html [hunk 3]
```diff
@@ -1367,14 +1367,14 @@ <h2>
...
+(for example, to assist with key recovery as discussed in Section
 <a href="#key-revocation-and-recovery"></a>) without enabling them to prove
-control (and thus be able to impersonate the controllers).
+control (and thus be able to impersonate the <a>DID controller</a>).
```
Comment (TallTed) [inline index.html]:
```suggestion
control (and thus be able to impersonate the <a>DID controller(s)</a>).
```

File: index.html [hunk 4]
```diff
@@ -2641,7 +2641,8 @@ <h1>
     <p>
 It is critically important to apply the principles of Privacy by Design
 to all aspects of decentralized identifier architecture, because DIDs
-and DID Documents are &mdash; by design &mdash; administered directly by their controllers.
+and DID Documents are &mdash; by design &mdash; administered directly by
+the <a>DID controller</a>.
```
Comment (TallTed) [inline index.html]:
```suggestion
the <a>DID controller(s)</a>.
```

Comment (TallTed) [reply index.html]: Github is doing something weird. The `<span ... /span>` markup doesn't belong; only the embedded `(s)`

**Comment by jandrieu** [comment]
First, note that this intersects with #2 but that shouldn't keep this from merging.

Second, I'm not sure how the ReSpec rules work, but will Ted's proposed <a>controller(s)</a> still link to the controller definition?

It might be more effective to use the indefinite article "a" instead of the definite article "the", as in 
> Authentication is the mechanism by which a DID controller can cryptographically prove that they are associated with that DID

instead of 
> Authentication is the mechanism by which the DID controller can cryptographically prove that they are associated with that DID

I believe that shift will also address the plurality question, as we no longer need to specify the cardinality of controller(s).

Original
> control (and thus be able to impersonate the <a>DID controller</a>).

Ted's suggestion
> control (and thus be able to impersonate the <a>DID controller(s)</a>).

My suggestion
> control (and thus be able to impersonate a <a>DID controller</a>).

**Comment by rhiaro** [comment]
> Second, I'm not sure how the ReSpec rules work, but will Ted's proposed controller(s) still link to the controller definition?

L75 of terms.html should be updated to add `controller(s)` to the matching terms so the link will work.

> It might be more effective to use the indefinite article "a" instead of the definite article "the"

Potential ambiguity where "a DID controller" could mean the controller of any DID when we want to mean the controller of the specific DID in question. Only likely to be called out by someone deliberately misunderstanding, so might be worth the tradeoff.

**Comment by TallTed** [comment]
Unfortunately, in addition to @rhiaro's point about the potential ambiguity, `a` doesn't make plain that multiple controllers are possible and permitted. The tweak to terms.html is tiny, is likely to be useful in other places, and is now in PR #83.

**Comment by jandrieu** [comment]
Ok. This reads fine to me, modulo the fixes we'll need to both the authentication section (controller != authenticator) and the issue of what do we call the parties involve. Let's get this in.

**Comment by dlongley** [comment]
> Ok. This reads fine to me, modulo the fixes we'll need to both the authentication section (controller != authenticator) and the issue of what do we call the parties involve. Let's get this in.

Yes, once this PR is in I plan on writing up a PR that should help address `controller != authenticator` to resolve issues #2 and #22.

**Comment by msporny** [comment]
Applied suggestions from @TallTed and @rhiaro which seem to address concerns raised by @peacekeeper and @jandrieu. I think we have consensus, editorial, merging.

**Pull request merged**